src/E225B0.c

```diff
@@ -1,7 +1,186 @@
 #include "common.h"
 
-INCLUDE_ASM(s32, "E225B0", func_802B7000_E225B0);
 
-INCLUDE_ASM(s32, "E225B0", func_802B72C0_E22870);
+typedef struct struct802B7C78 {
+    /* 0x00 */ f32 x;
+    /* 0x04 */ f32 y;
+    /* 0x08 */ f32 z;
+    /* 0x0C */ f32 scale;
+    /* 0x10 */ s32 unk_10;
+    /* 0x14 */ s32 unk_14;
+    /* 0x18 */ s32 unk_18;
+    /* 0x1C */ s32 unk_1C;
+    /* 0x20 */ s32 unk_20;
+    /* 0x24 */ s32 unk_24;
+    /* 0x28 */ s32 unk_28;
+} struct802B7C78;
+
+typedef struct UnknownCommand {
+    /* 0x00 */ Matrix4f* unk_00;
+    /* 0x04 */ Matrix4f* unk_04;
+    /* 0x08 */ s16 unk_08;
+    /* 0x0A */ s16 unk_0A;
+    /* 0x0C */ s16 unk_0C;
+    /* 0x0E */ s16 unk_0E;
+    /* 0x10 */ u8 unk_10;
+} UnknownCommand;
+
+extern Matrix4f D_802B7580_E22B30;
+extern Matrix4f D_802B7BA0_E23150;
+extern Matrix4f D_802B7BC0_E23170;
+extern Matrix4f D_802B7BE0_E23190;
+extern Matrix4f D_802B7C00_E231B0;
+
+extern void (*D_8010C93C)(void);
+extern struct802B7C78* D_802B7C78_E23228;
+extern struct8015A578 D_8015A578;
+
+void func_8013AA9C(s32, s32, s32, s32, s32, s32, s32);
+void func_8013AF70(s32, UnknownCommand*, s32, Matrix4f*);
+void func_802B735C_E2290C(void);
+
+void func_802B7000_E225B0(void) {
+    Matrix4f matrix1;
+    Matrix4f matrix2;
+    UnknownCommand command;
+
+    Gfx* oldMasterGfxPos;
+    f32 scale;
+    s32 temp;
+    s32 matrixIndex;
+    struct802B7C78* localD_802B7C78_E23228;
+    u16 oldMatrixListPos;
+
+    if (gPlayerStatus.animFlags & 0x100) {
+        guScaleF(matrix1, D_802B7C78_E23228->scale, D_802B7C78_E23228->scale,
+            D_802B7C78_E23228->scale);
+        guRotateF(matrix2, -gCameras[gCurrentCameraID].currentYaw, 0.0f, 1.0f, 0.0f);
+        guMtxCatF(matrix1, matrix2, matrix1);
+        guTranslateF(matrix2, D_802B7C78_E23228->x, D_802B7C78_E23228->y, D_802B7C78_E23228->z);
+        guMtxCatF(matrix1, matrix2, matrix2);
+        guMtxF2L(matrix2, (Mtx*) &gDisplayContext->matrixStack[gMatrixListPos]);
+
+        gSPMatrix(gMasterGfxPos++, &gDisplayContext->matrixStack[gMatrixListPos++], 3);
+        gSPDisplayList(gMasterGfxPos++, (u32) &D_802B7C00_E231B0);
```

Comment:
This cast is weird; shouldn't `D_802B7C00_E231B0` be a `Gfx[]`? Then you can reference it like
```suggestion
        gSPDisplayList(gMasterGfxPos++, D_802B7C00_E231B0);
```